### DISCUSSION TITLE
More simplified IEnumerable<KeyValuePair<TKey,TValue>> parameter?

### DISCUSSION THREAD

**Discussion by Flithor** [dotnet/roslyn#62815] [Ideas]
As we know, If we have a method with `Dictionary` or `KeyValuePair` paramter, and use them by hard code sequence, we need:
```
Void Method(Dictionary<int, string> dic) { }
//...
Method(new Dictionary<int, string>{ { 1, "V1" }, { 2, "V2" }, { 3, "V3" } });
```
Consider omitting the part of "new Dictionary<int, string>"?
```
Method({ { 1, "V1" }, { 2, "V2" }, { 3, "V3" } });
//Or
Method(new<int, string>{ { 1, "V1" }, { 2, "V2" }, { 3, "V3" } });
```
I'm not sure if it's ambiguous in some case.

---------------
I know I can use `ValueTuple` to make it a bit simpler:
```
Void Method((int, string)[] tuples) => Method(tuples.ToDictionary(t=>t.Item1, t=>t.Item2));
Void Method(Dictionary<int, string> dic) { }
//...
Method(new[] { (1, "V1"), (2, "V2"), (3, "V3") });
```
But it seems a little strange in this way.

**Comment by ufcpp** [comment]
You can write `Method(new() { { 1, "V1" }, { 2, "V2" }, { 3, "V3" } });`.

**Reply by Flithor** [reply]
When it added?

**Reply by ufcpp** [reply]
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new

**Reply by Flithor** [reply]
Opps, not works on .net core 3.1, but thanks.
